helper/tests/helper-tests-allavro/src/test/resources/allavro/RecordWithDefaultUnionFieldCorrect_reader.avsc

```diff
@@ -0,0 +1,27 @@
+{
+  "type": "record",
+  "name": "SchemaWithRecordDefault",
+  "fields": [
+    {
+      "name": "mainField",
+      "type": {
+        "type": "record",
+        "name": "InnerRecord",
+        "fields": [
+          {
+            "name": "f1",
+            "type": [
+              "string",
+              "int",
+              "null"
+            ]
+          }
+        ]
+      },
+      "default": {
+        "f1": "default",
+        "f2": "\u0000"
```

Comment:
there is no f2 field in the schema above??

Reply:
ah, good catch. removed